script/configure-web-proxy

```diff
@@ -36,11 +42,19 @@ sed -i -e 's/^.*httpsonly.*$/httpsonly = 0/g' /etc/openqa/openqa.ini
 if [[ $web_proxy == "nginx" ]]; then
     echo "Setting up nginx"
     sed "s/openqa.example.com/$(hostname)/" /etc/nginx/vhosts.d/openqa.conf.template > /etc/nginx/vhosts.d/openqa.conf
+    # IPv6 uses [::]:80 syntax instead of just a port
+    if [[ -n "$web_port" ]]; then
+        sed -Ei "s@(.*listen.*)80@\1${web_port}@g" /etc/nginx/vhosts.d/openqa.conf /etc/nginx/nginx.conf
+    fi
     sed -i -e "s/\(^[^#]*server_name  localhost;\)/#\1/" /etc/nginx/nginx.conf
 elif [[ $web_proxy == "apache" || $web_proxy == "apache2" ]]; then
     echo "Setting up apache"
     for i in headers proxy proxy_http proxy_wstunnel rewrite; do a2enmod $i; done
     sed "s/#ServerName.*$/ServerName $(hostname)/" /etc/apache2/vhosts.d/openqa.conf.template > /etc/apache2/vhosts.d/openqa.conf
+    test -n "$web_port" && sed -i "s/^Listen.*$/Listen $web_port/" /etc/apache2/listen.conf
+    if [[ -n "$web_port" ]]; then
+        sed -i "s/^Listen.*$/Listen $web_port/" /etc/apache2/listen.conf
+    fi
```

Comment:
You can also turn the condition other way around:
```suggestion
    test -z "$web_port" || sed -i "s/^Listen.*$/Listen $web_port/" /etc/apache2/listen.conf
```

Nevertheless, you do not want to have it twice here.

Reply:
![videoframe_1755](https://github.com/user-attachments/assets/fb9a41ba-6ffa-4a02-a8f6-e8e6cec113a2)